### PR TITLE
Update koa@2

### PR DESCRIPTION
**Pull request by kurayama**
Updates the project to be compatible with koa2.

- [x] Update koa@2.0.1
- [x] Bump dependencies
- [x] Change `mocha` to `jest`
- [x] Remove jscs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules
+coverage
```
Comment (ricardogama) [inline .gitignore]: Sort.

File: package.json [hunk 2]
```diff
@@ -4,14 +4,11 @@
...
-  "main": "dist/src",
   "repository": "git@github.com:seegno/koa-pagination",
   "scripts": {
     "changelog": "github_changelog_generator --no-issues --header-label='# Changelog' --future-release=v$npm_config_future_release && sed -i '' -e :a -e '$d;N;2,4ba' -e 'P;D' CHANGELOG.md",
-    "lint": "eslint src test && jscs src test",
-    "prepublish": "npm run transpile",
-    "test": "NODE_ENV=test mocha $npm_package_options_mocha",
-    "transpile": "rm -rf dist/* && babel src --out-dir dist/src",
+    "lint": "eslint index.js errors test",
+    "test": "jest index.js errors test --coverage",
```
Comment (ruimarinho) [inline package.json]: `jest --coverage --verbose`?

File: package.json [hunk 3]
```diff
@@ -23,29 +20,15 @@
...
-    "eslint": "^2.10.2",
-    "eslint-config-seegno": "^4.0.0",
-    "eslint-plugin-babel": "^3.2.0",
-    "eslint-plugin-sort-class-members": "^1.0.1",
-    "jscs": "^2.11.0",
-    "jscs-config-seegno": "^2.0.0",
-    "koa": "^1.2.0",
-    "mocha": "^2.4.5",
+    "eslint": "^3.16.1",
+    "eslint-config-seegno": "^8.0.1",
```
Comment (ruimarinho) [inline package.json]: 9.0.0

File: package.json [hunk 4]
```diff
@@ -4,48 +4,31 @@
...
   "author": "Seegno",
-  "main": "dist/src",
   "repository": "git@github.com:seegno/koa-pagination",
   "scripts": {
     "changelog": "github_changelog_generator --no-issues --header-label='# Changelog' --future-release=v$npm_config_future_release && sed -i '' -e :a -e '$d;N;2,4ba' -e 'P;D' CHANGELOG.md",
-    "lint": "eslint src test && jscs src test",
-    "prepublish": "npm run transpile",
-    "test": "NODE_ENV=test mocha $npm_package_options_mocha",
-    "transpile": "rm -rf dist/* && babel src --out-dir dist/src",
+    "lint": "eslint index.js errors test",
```
Comment (ruimarinho) [inline package.json]: Don't forget node env please.

Comment (kurayama) [inline package.json]: Done.

**Comment by ruimarinho** [comment]
Can you rebase this due to changes in README.md?

**Pull request merged**